cryptoservice/crypto_service.go

```diff
@@ -24,16 +25,16 @@ const (
 type CryptoService struct {
 	gun        string
 	passphrase string
-	keyStore   *trustmanager.KeyFileStore
+	keyStore   trustmanager.KeyStore
 }
 
 // NewCryptoService returns an instance of CryptoService
-func NewCryptoService(gun string, keyStore *trustmanager.KeyFileStore, passphrase string) *CryptoService {
+func NewCryptoService(gun string, keyStore trustmanager.KeyStore, passphrase string) *CryptoService {
 	return &CryptoService{gun: gun, keyStore: keyStore, passphrase: passphrase}
 }
 
 // Create is used to generate keys for targets, snapshots and timestamps
-func (ccs *CryptoService) Create(role string, algorithm data.KeyAlgorithm) (*data.PublicKey, error) {
+func (ccs *CryptoService) Create(role string, algorithm data.KeyAlgorithm) (data.Key, error) {
```

Comment:
I thought about this a bit, and I think we should change things so that we have separate data.PublicKey and data.PrivateKey interfaces. I'll work on this if I have time.